src/app/backend/handler/apihandler.go

```diff
@@ -996,6 +1008,107 @@ func (apiHandler *APIHandler) handleGetPods(
 	response.WriteHeaderAndEntity(http.StatusCreated, result)
 }
 
+var upgrader = websocket.Upgrader{}
+
+// WebSocketIO implements the Reader and Writer interfaces to support reading
+// and writing to a websocket from a remote command.
+type WebSocketIO struct {
+	conn *websocket.Conn
+}
+
+// Read reads messages from the client browser websocket and copies the given
+// messages to the given read buffer to be sent to the stdin of the command
+// running in the container.
+func (t WebSocketIO) Read(p []byte) (int, error) {
+	messageType, m, err := t.conn.ReadMessage()
+	if err != nil || messageType != websocket.TextMessage {
+		return 0, err
+	}
+
+	// TODO: Handle case where len(m) > len(p)? This seems not to happen in practice.
+	for i := range m {
+		p[i] = m[i]
+	}
+	return len(m), nil
+}
+
+// Write is called by the remotecommand Executor stream and writes the given
+// data to the client websocket.
+func (t WebSocketIO) Write(p []byte) (int, error) {
+	// TODO: Should binary messages be used? xterm.js seems to send text based messages.
```

Comment:
we send/receive binary in container-terminal, we convert in the browser https://github.com/kubernetes-ui/container-terminal/blob/master/container-terminal.js#L31